feat(examples): add AI-powered search example

denis-shvets opened the pull request:
## 🎯 Changes

### Quote from my article
> Let your users speak: Seamless app interaction through Natural Language 
>
> Crafting effective search interfaces often presents a significant challenge. You rely on obligatory calendars for date selection and an abundance of dropdowns to manage parameters like status or region. Despite these efforts, users frequently encounter difficulties in efficiently locating information or navigating to the desired section.
>
> What if we could offer a mechanism to bypass the friction and allow users to communicate their needs directly?
>
> 🔹 SV: "Visa mig alla utgångna ordrar från förra sommaren" (Show me all expired orders from last summer.)
> 🔹 EN: "Delayed air shipments to BR over 5kg this week"
> 🔹 EN: "I'm looking for red waterproof Nike shoes, size 44, under 100"
>
> Introducing Natural Language Search as a solution to this challenge. 

### Commit message
- Introduces a comprehensive React example demonstrating natural language search capabilities
- Users can query merchant data (orders, disputes, settlements) using conversational language like "show me orders from last week"
- AI converts natural language prompts into structured search parameters with proper filtering and date ranges
- Includes full UI with data tables, filters, and responsive design using Tailwind CSS
- Leverages TanStack Start, TanStack Router, TanStack AI

### Preview
<img width="2648" height="2298" alt="orders-preview" src="https://github.com/user-attachments/assets/50657fb6-09d8-40ff-bedc-8062f6908534" />

## Next steps
- At the moment, the filters and tables rely on the original implementation. The next phase will be migrating them to TanStack Form, TanStack Table, and TanStack Virtual(maybe?).
- The data is also hardcoded for now. The plan is to transition to TanStack Query or TanStack DB.
- Clean up the leftover TanStack Start artifacts.

## ✅ Checklist

- 🤷 I have followed the steps in the [Contributing guide](https://github.com/TanStack/ai/blob/main/CONTRIBUTING.md). - The file does not exist. 🚨 
- [x] I have tested this code locally with `pnpm run test:pr`

## 🚀 Release Impact

This change introduces only the new example.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a comprehensive TypeScript React search example project featuring natural language search powered by AI integration
  * Implemented voice input support for hands-free search queries
  * Added interactive data management interfaces for disputes, orders, and settlements with filtering capabilities
  * Created reusable UI components including filters, date pickers, tables, and selectors
  * Integrated TanStack Router for client-side navigation and TanStack React Query for server state management
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

denis-shvets committed:
feat(examples): add AI-powered search example

- Introduces a comprehensive React example demonstrating natural language search capabilities
- Users can query merchant data (orders, disputes, settlements) using conversational language like "show me orders from last week"
- AI converts natural language prompts into structured search parameters with proper filtering and date ranges
- Includes full UI with data tables, filters, and responsive design using Tailwind CSS
- Leverages TanStack Start, TanStack Router, TanStack AI

denis-shvets committed:
feat(examples/ts-react-search): add navigation component to hero section

- Added a new Navigation component with links to Home, Orders, Disputes, and Settlements pages
- Integrated navigation into the hero section for improved user experience

denis-shvets committed:
refactor(navigation): simplify route references in Navigation component

- Replace imported route objects with hardcoded string paths
- Remove unused route imports

denis-shvets committed:
refactor(routes): restructure API search route into directory

- Moved api.search.ts to api/search.ts for better organization
- Updated route tree imports to reflect new file structure
- Maintains existing functionality while improving code organization

jherr commented:
We should hold on this example until we have a structured output generation function.

denis-shvets committed:
feat(examples): integrate TanStack DB for client-side data management

- Replace server functions with TanStack DB collections and live queries
- Add @tanstack/react-db, @tanstack/query-db-collection, and related packages
- Implement disputes, orders, and settlements collections with Zod validation
- Create useLiveQuery hooks for reactive data filtering and searching
- Update components to use client-side collections instead of server functions

denis-shvets committed:
feat(examples): update search API to use server-sent events streaming

- Migrated from `toStreamResponse` to `toServerSentEventsResponse` for improved streaming
- Updated OpenAI adapter to use `openaiText` with model specification
- Updated multiple dependencies including React, TanStack Router, and TailwindCSS

denis-shvets committed:
refactor(search): migrate search API from streaming to synchronous with structured validation

- Replaced Server-Sent Events with React Query mutation for search requests
- Added Zod schema validation for structured output in search API
- Updated search component to handle JSON responses instead of streaming
- Improved error handling and type safety for search parameters

denis-shvets committed:
refactor(search): extract search mutation logic into reusable hook

- Moved search API mutation logic from Search component into dedicated useSearchMutation hook
- Improves code reusability and separation of concerns
- Enables search functionality to be used across multiple components
- Reduces code duplication and improves maintainability

AlemTuzlak commented:
Hey Denis, mind just merging latewt main and making sure it works and I can merge this, thank you so much for this and sorry for the wait!

denis-shvets commented:
> Hey Denis, mind just merging latewt main and making sure it works and I can merge this, thank you so much for this and sorry for the wait!

@AlemTuzlak no worries, everyone has a lot of work. I will update it this week and let you know.

denis-shvets committed:
build(ts-react-search): bump TanStack Router stack and refresh lockfile

- Align the ts-react-search example with newer @tanstack/react-router, react-start, router-plugin, and devtools packages so it stays compatible with current TanStack releases
- Regenerated route tree types so the layout route reports fullPath as '/' instead of an empty string, matching the updated router codegen
- Updated the workspace lockfile so installs resolve consistently with the new dependency graph

denis-shvets committed:
feat(ts-react-search): switch AI search adapter from OpenAI to Groq

- Example depends on @tanstack/ai-groq and uses groqText with openai/gpt-oss-20b
- Server checks GROQ_API_KEY instead of OPENAI_API_KEY
- Output schema is wrapped with toGroqCompatibleSchema so Groq accepts JSON Schema unions (additionalProperties on anyOf)

autofix-ci[bot] committed:
ci: apply automated fixes

denis-shvets commented:
Hey @AlemTuzlak

I do not currently have access to OpenAI or Anthropic API keys, so I updated the example to use the free Groq adapter instead. Since it does not support unions, I added a `toGroqCompatibleSchema` utility as a workaround. This may be a limitation or bug in the Groq adapter. I'll investigate further later.

So far, everything is working as expected on the Groq free-tier model.
